numpy/lib/function_base.py

```diff
@@ -393,9 +396,11 @@ def average(a, axis=None, weights=None, returned=False):
         wgt = np.asanyarray(weights)
 
         if issubclass(a.dtype.type, (np.integer, np.bool_)):
-            result_dtype = np.result_type(a.dtype, wgt.dtype, 'f8')
+            result_dtype = np.result_type.__skip_array_function__(
+                a.dtype, wgt.dtype, 'f8')
         else:
-            result_dtype = np.result_type(a.dtype, wgt.dtype)
+            result_dtype = np.result_type.__skip_array_function__(
```

Comment:
Ah, wow did not realize we even dispatch for things such as `.dtype` attribute (just me realizing the extend of this :)).

Reply:
It's because `np.result_type` works on arrays/array-likes, too, e.g., you can write `np.result_type([1])`.

Reply:
Right... comment was really only, because I was a bit curious how much we could try to seperate out dtypes and containers.

Reply:
If we want to do that separation, we might want to not allow overloads of result_type for container types...